### PR TITLE
Fix OS_NAME dep warn

### LINKED ISSUE #223: OS_NAME doesn't exist anymore (0.5-master)
As of https://github.com/JuliaLang/julia/commit/694413abca84b1745afe000f05a1676e899e6a43 (0.5-dev) OS_NAME doesn't exist anymore what leads to the following error (and a lot of deprecated warnings):
```
ERROR: LoadError: LoadError: UndefVarError: OS_NAME not defined
 in include_from_node1(::String) at .\loading.jl:426 (repeats 2 times)
 in macro expansion; at .\none:2 [inlined]
 in anonymous at .\<missing>:?
 in eval(::Module, ::Any) at .\boot.jl:225
 in process_options(::Base.JLOptions) at .\client.jl:243
 in _start() at .\client.jl:322
while loading C:\Users\necka\.julia\v0.5\BinDeps\src\dependencies.jl, in expression starting on line 606
while loading C:\Users\necka\.julia\v0.5\BinDeps\src\BinDeps.jl, in expression starting on line 513
ERROR: Failed to precompile BinDeps to C:\Users\necka\.julia\lib\v0.5\BinDeps.ji
 in compilecache(::String) at .\loading.jl:505
 in require(::Symbol) at .\loading.jl:364
 in eval(::Module, ::Any) at .\boot.jl:225
 in macro expansion at .\REPL.jl:92 [inlined]
 in (::Base.REPL.##1#2{Base.REPL.REPLBackend})() at .\event.jl:46
```

### PR DESCRIPTION
**Pull request by ranjanan**
`OS_NAME` to `Compat.KERNEL`

Fixes #223

### CODE REVIEW

**Comment by tkelman** [comment]
see recent discussions on this. these have different values and this is part of the API, so cannot be a dropin replacement. you hace to translate to the old values, `:Windows` instead of `:NT`

File: src/dependencies.jl [hunk 1]
```diff
@@ -619,7 +619,8 @@ end
 
 function applicable(dep::LibraryDependency)
     if haskey(dep.properties,:os)
-        if (dep.properties[:os] != OS_NAME && dep.properties[:os] != :Unix) || (dep.properties[:os] == :Unix && !is_unix())
+        if (dep.properties[:os] != (@compat @static is_unix() ? Compat.KERNEL : :NT)
```
Comment (tkelman) [inline src/dependencies.jl]: wrong way around. `:NT` should not appear in the BinDeps API at all, we only want to compare against `:Windows`

**Comment by tkelman** [comment]
the simplest way to do this is going to be
```
OSNAME = is_windows() ? :Windows : Compat.KERNEL
```
then use this variable instead of `OS_NAME`

File: src/dependencies.jl (hunk 1 above)
Comment (ranjanan) [inline src/dependencies.jl]: Oh yeah, :Windows, not :NT.

File: src/dependencies.jl [hunk 2]
```diff
@@ -1,5 +1,6 @@
 # This is the high level interface for building dependencies using the declarative BinDeps Interface
 import Base: show
+const OSNAME = @static is_windows() ? :Windows : Compat.KERNEL
```
Comment (tkelman) [inline src/dependencies.jl]: shouldn't need `@static` here

File: src/dependencies.jl [hunk 3]
```diff
@@ -644,7 +645,8 @@ function can_provide(p,opts,dep)
 end
 
 function can_provide(p::PackageManager,opts,dep)
-    if p === nothing || (haskey(opts,:os) && opts[:os] != OS_NAME && (opts[:os] != :Unix || !is_unix()))
+    if p === nothing || (haskey(opts,:os) && opts[:os] != OSNAME && 
+        (opts[:os] != :Unix || !is_unix()))
```
Comment (tkelman) [inline src/dependencies.jl]: leave this as a single line the way it was

**Pull request merged**